lib/src/test/java/io/ably/lib/test/realtime/RealtimeConnectFailTest.java

```diff
@@ -546,4 +555,198 @@ public void onConnectionStateChanged(ConnectionStateChange state) {
+                        System.out.println("onConnectionStateChanged retry count is: " + retryCount.get());
+                        if (retryCount.get() > 4) {
+                            System.out.println("onConnectionStateChanged retry is successful and done!");
+                            return;
+                        }
+                        retryCount.incrementAndGet();
+                        retryValues.add(state.retryIn);
+                    }
+                }
+            });
+
+            int waitAtMost = 5 * 10; //5 seconds * 10 times per second
+            int waitCount = 0;
+            while (retryCount.get() < 4 && waitCount < waitAtMost) {
+                try {
+                    Thread.sleep(100);
+                    waitCount++;
+                } catch (InterruptedException e) {
+                    fail(e.getMessage());
+                }
+            }
+            System.out.println("wait done in: " + (waitCount / 10) + " seconds");
+
+            assertTrue("Disconnect retry was not finished, count was: " + retryCount.get(), retryCount.get() >= 4);
+
+            //check for all received retry times in onConnectionStateChanged callback
+            System.out.println("------------------------------------------------------------");
+            for (int i = 0; i < retryValues.size(); i++) {
+                long retryTime = retryValues.get(i);
+                long higherRange = disconnectedRetryTimeout + Math.min(i, 3) * 50L;
+                double lowerRange = 0.6 * disconnectedRetryTimeout + Math.min(i, 3) * 50L;
+
+                System.out.println("higher range: " + higherRange + " - lower range: " + lowerRange + " | checked value: " + retryTime);
+
+                assertTrue("retry time higher range for count " + i + " is not in valid: " + retryTime + " expected: " + higherRange,
+                    retryTime < higherRange);
+                assertTrue("retry time lower range for count " + i + " is not in valid: " + retryTime + " expected: " + lowerRange,
+                    retryTime > lowerRange);
+            }
+            System.out.println("------------------------------------------------------------");
+        } catch (AblyException e) {
+            e.printStackTrace();
+            fail("init0: Unexpected exception instantiating library");
+        } finally {
+            Defaults.TIMEOUT_DISCONNECT = oldDisconnectTimeout;
+            if (ably != null)
+                ably.close();
+        }
+    }
+
+    /**
+     * Connect and check if timer time is jittered
+     * Spec: RTB1
+     */
+    @Test
+    public void disconnect_retry_channel_timeout_jitter() {
+        long oldRealtimeTimeout = Defaults.realtimeRequestTimeout;
+        int channelRetryTimeout = 150;
+        /* Reduce timeout for test to run faster */
+        Defaults.realtimeRequestTimeout = channelRetryTimeout;
```

Comment:
This looks strange to me - you're setting the `realtimeRequestTimeout` to a value which you're calling `channelRetryTimeout`, which should be two separate timeouts. And is there a reason not to use the `channelRetryTimeout` client option here?

Reply:
Setting channelRetryTimeout did not work alone and from [other test](https://github.com/ably/ably-java/blob/main/lib/src/test/java/io/ably/lib/test/realtime/RealtimeChannelTest.java#L1726-L1801) I saw that setting the default value does the trick.

Reply:
I am not sure if we should use Random seed as it will give us a static value when we expect a random range.
Random range helped me find the issue in the core implementation as well.